tests/plugins/emoji/basic.js

```diff
@@ -177,16 +167,21 @@
 					editable = editor.editable();
 
 				bot.setHtmlWithSelection( '<p>foo</p><p>:bug^</p>' );
+				editable.fire( 'keyup', new CKEDITOR.dom.event( {} ) );
+				objectAssert.areEqual( { id: ':bug:', symbol: '🐛' }, autocomplete.model.data[ 0 ], 'Emoji result contains wrong result' );
+			} );
+		},
 
-				// Delay assertions because of autocomplete throttle.
-				setTimeout( function() {
-					resume( function() {
-						objectAssert.areEqual( { id: ':bug:', symbol: '🐛' }, autocomplete.model.data[ 0 ], 'Emoji result contains wrong result' );
-					} );
-				}, 50 );
+		// (#2394)
+		'test emoji correctly matches repeated keywords': function( editor, bot ) {
+			emojiTools.runAfterInstanceReady( editor, bot, function( editor, bot ) {
+				var autocomplete = editor._.emoji.autocomplete;
 
-				editable.fire( 'keyup', new CKEDITOR.dom.event( {} ) );
-				wait();
+				bot.setHtmlWithSelection( '<p>foo :collision :collision^</p>' );
+				editor.editable().fire( 'keyup', new CKEDITOR.dom.event( {} ) );
+				assert.areSame( ':collision', autocomplete.model.query, 'Model keeps wrong querry.' );
```

Comment:
There should be no dots at the end of assert msg. I can see that you might have take it from other asserts in this suite.

What's interesting is that `objectAssert`s have correct form 🙂

Reply:
typo querry -> query